phoenix-core/src/it/java/org/apache/phoenix/end2end/UpsertSelectIT.java

```diff
@@ -101,145 +122,163 @@ private void testUpsertSelect(boolean createIndex, boolean saltTable) throws Exc
-        assertEquals("3", rs.getString(2));
-        assertEquals(A_VALUE, rs.getString(3));
-        
-        assertTrue (rs.next());
-        assertEquals("00A", rs.getString(1));
-        assertEquals("4", rs.getString(2));
-        assertEquals(A_VALUE, rs.getString(3));
+        // Trigger multiple batches
+        props.setProperty(UPSERT_BATCH_SIZE_ATTRIB, Integer.toString(3));
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            conn.setAutoCommit(true);
+            String upsert = "UPSERT INTO " + customEntityTable +
+                    "(custom_entity_data_id, key_prefix, organization_id, created_by) " +
+                    "SELECT substr(entity_id, 4), substr(entity_id, 1, 3), organization_id, " +
+                    "a_string  FROM " + aTable + " WHERE ?=a_string";
+            if (createIndex) { // Confirm index is used
+                try (PreparedStatement upsertStmt =
+                        conn.prepareStatement("EXPLAIN " + upsert)) {
+                    upsertStmt.setString(1, tenantId);
+                    ResultSet ers = upsertStmt.executeQuery();
+                    assertTrue(ers.next());
+                    String explainPlan = QueryUtil.getExplainPlan(ers);
+                    assertTrue(explainPlan.contains(" SCAN OVER " + indexName));
+                }
+            }
+
+            try (PreparedStatement upsertStmt = conn.prepareStatement(upsert)) {
+                upsertStmt.setString(1, A_VALUE);
+                int rowsInserted = upsertStmt.executeUpdate();
+                assertEquals(4, rowsInserted);
+            }
+            conn.commit();
+        }
+
+        String query = "SELECT key_prefix, substr(custom_entity_data_id, 1, 1), created_by FROM " +
+                customEntityTable + " WHERE organization_id = ? ";
+        try (Connection conn = DriverManager.getConnection(getUrl(), props);
+                PreparedStatement statement = conn.prepareStatement(query)) {
+            statement.setString(1, tenantId);
+            ResultSet rs = statement.executeQuery();
+
+            assertTrue (rs.next());
+            assertEquals("00A", rs.getString(1));
+            assertEquals("1", rs.getString(2));
+            assertEquals(A_VALUE, rs.getString(3));
+
+            assertTrue (rs.next());
+            assertEquals("00A", rs.getString(1));
+            assertEquals("2", rs.getString(2));
+            assertEquals(A_VALUE, rs.getString(3));
+
+            assertTrue (rs.next());
+            assertEquals("00A", rs.getString(1));
+            assertEquals("3", rs.getString(2));
+            assertEquals(A_VALUE, rs.getString(3));
+
+            assertTrue (rs.next());
+            assertEquals("00A", rs.getString(1));
+            assertEquals("4", rs.getString(2));
+            assertEquals(A_VALUE, rs.getString(3));
 
-        assertFalse(rs.next());
-        conn.close();
+            assertFalse(rs.next());
```

Comment:
looks like this was a tab before and now a space (which is consistent) so valid diff imo